plutus-wallet-api/ledger/Ledger/Validation.hs

```diff
@@ -141,20 +127,28 @@ findData dsh PendingTx{pendingTxData=datas} = snd <$> find f datas
     where
         f (dsh', _) = dsh' == dsh
 
+{-# INLINABLE findDataHash #-}
+-- | Find the hash of a data value, if it is part of the pending transaction's
+--   hashes
+findDataHash :: DataValue -> PendingTx -> Maybe DataValueHash
```

Comment:
Maybe we can use `findDataHash` instead of a `hash :: DataValue -> DataValueHash` primitive (for oracles). Then you just have to supply the data that was signed by the oracle as part of the transaction's data values.